pkg/provider/provider.go

```diff
@@ -49,6 +54,27 @@ type Legacy struct {
 func NewProviderCreator(jwtSecret string, legacy Legacy) ProviderCreator {
 	return func(user, password string) (Provider, error) {
 		switch {
+		case viper.IsSet("authorization-lambda"):
```

Comment:
Don't read configuration from arbitrary places within the code. For now all config is parsed and transformed into domain models within `main.go`. Having no side effects also improves (unit) testability.

Reply:
Will do. I'm a complete novice with go, so I was trying to limit my changes as much as possible. But this should be very easy to do.